http/handler.go

```diff
@@ -164,15 +179,21 @@ func (h *handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	err = h.root.Call(req, re, h.env)
 	if err != nil {
 		re.SetError(err, cmdkit.ErrNormal)
+		otExt.Error.Set(rootSpan, true)
+		rootSpan.LogKV("error", err.Error())
 		return
 	}
 
 	d := re.(Doner)
 	select {
 	case <-d.Done():
 	case <-req.Context.Done():
-		log.Errorf("waiting for http.Responseemitter to close but then %s", req.Context.Err())
+		errMsg := "waiting for http.Responseemitter to close but then " + req.Context.Err().Error()
```

Comment:
Using `fmt.Sprintf` with `req.Context.Err()` would be a bit safer, though I don't think that `req.Context.Err()` should ever return nil.